### PR TITLE
Rewrite fty-alert-engine

### PR DESCRIPTION
**Pull request by jana-rapava**
This PR is part of alert refactoring.

### CODE REVIEW

File: project.xml [hunk 1]
```diff
@@ -41,6 +41,9 @@
         <option name = "use_checkout_explicit" value = "1" />
     </target>
 
+    <classfilename use-cxx = "true" source-extension = "cc" header-extension = "h"
```
Comment (perrettecl) [inline project.xml]: Line 15 in Travis section:
'option name = "dist" value = "trusty" '
'option name = "shadow_gcc" value = "2" '

Comment (perrettecl) [inline project.xml]: Enable gcc-4.9 may be needed for Travis (because of shm) =>  use-cxx-gcc-4-9 = "true"

File: project.xml [hunk 2]
```diff
@@ -150,25 +153,84 @@
...
+            test = "fty_common_logging_selftest" >
+
+            <use project = "log4cplus" header = "log4cplus/logger.h" test = "appender_test"
+                repository = "https://github.com/42ity/log4cplus.git"
+                release = "1.1.2-FTY-master"
+                />
+        </use>
+    </use>
+
     <use project = "fty_shm" libname = "libfty_shm" header="fty_shm.h" min_major = "1" test = "fty_shm_test"
```
Comment (perrettecl) [inline project.xml]: Format fty_shm  use section like the other (mentioning release="master")

File: README.md [hunk 3]
```diff
@@ -43,23 +41,76 @@ Rules loaded at start up are stored in the directory /var/lib/fty/fty-alert-engi
...
 ### Rule templates
 
-To be added.
+Rules are currently of types flexible, single, pattern and threshold
+UPPERCASEs are variables to be filled
+lowercases are fixed
+// comments are to put more details, not part of json format
+one|other means OR operator, provided either one or other applies
+... means more might follow
+{
```
Comment (boricj) [inline README.md]: Put this in a Markdown code block for pretty-formatting.

Comment (jana-rapava) [reply README.md]: We will address that in phase 2.

**Pull request merged**